### PR TITLE
Make AnalysisResult.fit optional

### LINKED ISSUE #916: test_experiment.py tests fail with "TypeError: __init__() missing 1 required positional argument: 'value'"
### Information

- **Qiskit IBMQ Provider version**: master (git hash fe5a07c7f8983128425b094f0d28daa01f9addec)
- **Python version**: 3.8.5
- **Operating system**: Ubuntu Focal 20.04

### What is the current behavior?

I got the test/ibmq/test_experiment.py tests running against the `staging` instance of the resultsdb service, here are the results:

https://gist.github.com/mriedem/2d7b3b9a4af230e1bc1fad8ff8a550d9

The main issue seems to be this:

```
======================================================================
ERROR: test_analysis_results_experiment_uuid (test.ibmq.test_experiment.TestExperiment)
Test filtering analysis results with experiment uuid.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/osboxes/ibmq/qiskit-ibmq-provider/test/ibmq/test_experiment.py", line 516, in test_analysis_results_experiment_uuid
    ref_result = self.provider.experiment.analysis_results()[0]
  File "/home/osboxes/ibmq/qiskit-ibmq-provider/qiskit/providers/ibmq/experiment/experimentservice.py", line 362, in analysis_results
    results.append(AnalysisResult.from_remote_data(result))
  File "/home/osboxes/ibmq/qiskit-ibmq-provider/qiskit/providers/ibmq/experiment/analysis_result.py", line 173, in from_remote_data
    obj = cls(experiment_uuid=remote_data['experiment_uuid'],
  File "/home/osboxes/ibmq/qiskit-ibmq-provider/qiskit/providers/ibmq/experiment/analysis_result.py", line 85, in __init__
    self.fit = fit  # type: ignore[assignment]
  File "/home/osboxes/ibmq/qiskit-ibmq-provider/qiskit/providers/ibmq/experiment/analysis_result.py", line 134, in fit
    fit_val = Fit(**fit_val)
TypeError: __init__() missing 1 required positional argument: 'value'
```

That's because the `AnalysisResults.fit` setter code assumes that `fit` will always have value but that changed in version 0.20.1 of the API to accommodate Ignis experiments which might not have a `fit` value.

There was one other error in those test results:

```
======================================================================
ERROR: test_update_analysis_result (test.ibmq.test_experiment.TestExperiment)
Test updating an analysis result.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/osboxes/ibmq/qiskit-ibmq-provider/qiskit/providers/ibmq/api/session.py", line 272, in request
    response.raise_for_status()
  File "/home/osboxes/ibmq/qiskit-ibmq-provider/.venv/lib/python3.8/site-packages/requests/models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://api-dev.quantum-computing.ibm.com/resultsdb/analysis_results/4ce3753a-cc01-5567-bbfe-de6d33601782

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/osboxes/ibmq/qiskit-ibmq-provider/test/ibmq/test_experiment.py", line 454, in test_update_analysis_result
    self.provider.experiment.update_analysis_result(new_result)
  File "/home/osboxes/ibmq/qiskit-ibmq-provider/qiskit/providers/ibmq/experiment/experimentservice.py", line 429, in update_analysis_result
    response = self._api_client.analysis_result_update(result.uuid, data)
  File "/home/osboxes/ibmq/qiskit-ibmq-provider/qiskit/providers/ibmq/api/clients/experiment.py", line 250, in analysis_result_update
    return self.base_api.analysis_result(result_id).update(new_data)
  File "/home/osboxes/ibmq/qiskit-ibmq-provider/qiskit/providers/ibmq/api/rest/analysis_result.py", line 52, in update
    return self.session.put(url, json=analysis_result).json()
  File "/home/osboxes/ibmq/qiskit-ibmq-provider/.venv/lib/python3.8/site-packages/requests/sessions.py", line 602, in put
    return self.request('PUT', url, data=data, **kwargs)
  File "/home/osboxes/ibmq/qiskit-ibmq-provider/qiskit/providers/ibmq/api/session.py", line 294, in request
    raise RequestsApiError(message, status_code) from ex
qiskit.providers.ibmq.api.exceptions.RequestsApiError: '400 Client Error: Bad Request for url: https://api-dev.quantum-computing.ibm.com/resultsdb/analysis_results/4ce3753a-cc01-5567-bbfe-de6d33601782. {"errors":["chisq is not a decimal"]}'
```

### Steps to reproduce the problem

Just run the test/ibmq/test_experiment.py tests.

### What is the expected behavior?

Tests should pass.

### Suggested solutions

- Handle `AnalysisResult.fit` being an empty dict.
- Fix the `chisq` value used in the `test_update_analysis_result` test

### PR DESCRIPTION
**Pull request by mriedem**
### Summary

The `fit` attribute is now optional in the service and
if specified it does not need `value` or `variance` values,
it's just a free-form dict and defaults to an empty dict.
As such, the special `Fit` class is removed here and tests
are updated.

To get tests to pass, `test_analysis_results_quality` is
skipped until issue #902 is fixed.

Also while in here, the temporary `hello_world.svg` plot
file is removed on cleanup of the test case.

### Details and comments

Closes #916

### CODE REVIEW

**Comment by mriedem** [comment]
I don't know why pylint failed here:

https://github.com/Qiskit/qiskit-ibmq-provider/pull/917/checks?check_run_id=2356370043#step:5:10

>test/ibmq/test_experiment.py:465:46: E1101: Class 'ResultQuality' has no 'BAD' member (no-member)

The `ResultQuality.BAD` enum value clearly exists:

https://github.com/Qiskit/qiskit-ibmq-provider/blob/79006fd04bcb098c7aa1876a548992d01850d25f/qiskit/providers/ibmq/experiment/constants.py#L51

@jyu00 any idea on why that might happen?

**Comment by mriedem** [comment]
> 
> 
> I don't know why pylint failed here:
> 
> https://github.com/Qiskit/qiskit-ibmq-provider/pull/917/checks?check_run_id=2356370043#step:5:10
> 
> > test/ibmq/test_experiment.py:465:46: E1101: Class 'ResultQuality' has no 'BAD' member (no-member)
> 
> The `ResultQuality.BAD` enum value clearly exists:
> 
> https://github.com/Qiskit/qiskit-ibmq-provider/blob/79006fd04bcb098c7aa1876a548992d01850d25f/qiskit/providers/ibmq/experiment/constants.py#L51
> 
> @jyu00 any idea on why that might happen?

Rebased, let's see if that resolves it.

**Comment by mriedem** [comment]
I don't know what's going on with the pylint failures here, can't reproduce. I wonder if there is a bad cache in the build?

File: test/ibmq/test_experiment.py [hunk 1]
```diff
@@ -406,8 +406,8 @@ def test_get_analysis_results(self):
         results = self.provider.experiment.analysis_results()
         for res in results:
             self.assertTrue(isinstance(res, AnalysisResult))
-            self.assertFalse(res.verified)
-            self.assertTrue(isinstance(res.fit, Fit))
+            self.assertIsInstance(res.verified, bool)
+            self.assertTrue(isinstance(res.fit, Dict))
```
Comment (jyu00) [inline test/ibmq/test_experiment.py:410]:
```suggestion
            self.assertIsInstance(res.fit, dict)
```

File: test/ibmq/test_experiment.py [hunk 2]
```diff
@@ -449,17 +449,20 @@ def test_update_analysis_result(self):
...
 
         rresult = self.provider.experiment.retrieve_analysis_result(new_result.uuid)
         sub_tests = [(new_result, 'local result'), (rresult, 'retrieved result')]
         for res, name in sub_tests:
             with self.subTest(sub_test_name=name):
-                self.assertEqual(res.fit.to_dict(), new_fit.to_dict())
+                self.assertEqual(res.fit, new_fit)
                 self.assertEqual(res.type, original_type)
-                self.assertEqual(res.quality, ResultQuality.HUMAN_BAD)
+                self.assertEqual(res.quality, ResultQuality.BAD)
```
Comment (jyu00) [inline test/ibmq/test_experiment.py:465]:
```suggestion
                self.assertEqual(res.quality, ResultQuality.BAD)  # pylint: disable=no-member
```

**Review by jyu00 — APPROVED** [review]

**Comment by jyu00** [comment]
The pylint error looks to be related to custom Enum class. [This issue](https://github.com/PyCQA/pylint/issues/533) is still open so we can only disable the check now

**Comment by mriedem** [comment]
> 
> 
> The pylint error looks to be related to custom Enum class. [This issue](https://github.com/PyCQA/pylint/issues/533) is still open so we can only disable the check now

Great, thanks.  I was thinking about working around that the same way, but I wasn't aware of the outstanding issue.

**Pull request merged**